### PR TITLE
Added UMD definitions

### PR DESCRIPTION
**Pull request by ealves-pt**

### CODE REVIEW

**Comment by ealves-pt** [comment]
Closed in favour of #277 that will include this changes.

File: Gruntfile.js [hunk 1]
```diff
@@ -26,8 +26,16 @@ module.exports = function(grunt) {
...
         banner: '<%= meta.banner %>' + '\n' +
-          '(function ( window, angular, undefined ) {' + '\n',
-        footer: '})( window, window.angular );'
+        "(function (factory) {\n" +
+        "  if (typeof define === 'function' && define.amd) {\n" +
+        "    define(['exports', 'angular'], factory);\n" +
+        "  } else if (typeof module === 'object' && typeof module.exports === 'object') {\n" +
+        "    factory(module.exports, require('angular'));\n" +
+        "  } else if (root.hasOwnProperty('angular')) {\n" +
+        "    factory((root.exports || (root.exports = {})), root.angular);\n" +
```
Comment (idiotWu) [inline Gruntfile.js]: Actually I don't understand why the module should be attached to `root.exports` object, this may cause conflicts like #288

Comment (ealves-pt) [inline Gruntfile.js]: In that case how do you purpose that I would define de module if no AMD nor Common definition are available?

I have followed more or less [this](http://davidbcalhoun.com/2014/what-is-amd-commonjs-and-umd/) approach.

Comment (idiotWu) [inline Gruntfile.js]: I think you should export it directly to the global object like:

```javascript
factory(this, this.angular)
```

Here's also an example of UMD definition which requires a return value of you module: https://github.com/umdjs/umd/blob/master/templates/returnExports.js

**Pull request closed without merging**